### DISCUSSION TITLE
[Feature Request] Allow use of variables in section settings’ "Template" field (like "Entry URI Format")

### DISCUSSION THREAD

**Discussion by benface** [craftcms/cms#8760] [Ideas]
When creating a Channel or Structure section, it would be useful if we could use variables in the "Template" field(s) below "Site Settings", just like we can in the "Entry URI Format" field(s). For instance, I would like to load a different template based on the entry's type, so I would do it like this:

![template](https://user-images.githubusercontent.com/1059139/71041906-459b3800-20f8-11ea-9cb9-8b768620c74a.jpg)

But right now I get the following error message when visiting the URL of an entry in that section:

> Template not found: pages/{type.handle}

I know I can achieve the same effect by pointing to a single template that includes a different template based on the `entry.type.handle`, but I would like to avoid that `{% include %}` if possible (not sure if it adds any overhead at all, but just from a clean code perspective).

Thanks for considering :)

**Comment by brandonkelly** [comment]
I’d do this right away, but rendering object templates like those is a little expensive, so not something you want to do on every request. We’d need to start prerendering the templates and storing them for each entry/site when entries are saved. We can look into that when we get to #2450.

**Comment by benface** [comment]
@brandonkelly Makes sense, thanks!
